client/components/Leaderboard.tsx

```diff
@@ -3,147 +3,38 @@ import { Trophy } from 'lucide-react';
-        },
-        {
-            rank: 3,
-            name: 'Team 4',
-            score: 650,
-            tests: [
-                'pass',
-                'pass',
-                'pass',
-                'in-progress',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-            ],
-        },
-        {
-            rank: 4,
-            name: 'Team 5',
-            score: 540,
-            tests: [
-                'pass',
-                'pass',
-                'in-progress',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-            ],
-        },
-        {
-            rank: 5,
-            name: 'Team 6',
-            score: 530,
-            tests: [
-                'pass',
-                'pass',
-                'in-progress',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-            ],
-        },
-        {
-            rank: 6,
-            name: 'Team 7',
-            score: 520,
-            tests: [
-                'pass',
-                'pass',
-                'in-progress',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-            ],
-        },
-        {
-            rank: 7,
-            name: 'Team 8',
-            score: 510,
-            tests: [
-                'pass',
-                'pass',
-                'in-progress',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-            ],
-        },
-        {
-            rank: 8,
-            name: 'Team 9',
-            score: 500,
-            tests: [
-                'pass',
-                'pass',
-                'in-progress',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-                'not-attempted',
-            ],
-        },
-    ];
+    const leadboardData = useLeaderboard();
+
+    console.log(leadboardData);
 
     return (
         <div className="flex flex-col items-center gap-4">
-            {data.map((player) => (
+            {leadboardData.map((player, index) => (
                 <Card
-                    key={player.rank}
+                    key={player.username}
                     className="flex w-1/2 min-w-[600px] flex-row p-6 text-xl shadow-md"
                 >
                     <div className="flex w-1/3 flex-row items-center gap-2">
-                        <b>{player.name}</b>
-                        {player.rank < 3 && (
-                            <span className={trophyColor(player.rank)}>
+                        <b>{player.username}</b>
+                        {index + 1 < 3 && (
```

Comment:
Go ahead and fix this:
```suggestion
                        {index < 3 && (
```